Decred Wallet/Custom Views/FloatingLabelTextInput.swift

```diff
@@ -0,0 +1,166 @@
+    var activeBorderColor: UIColor = UIColor.appColors.decredBlue
+    let inactiveBorderColor: UIColor = UIColor.appColors.lightGray
+    
+    let cornerRadius: CGFloat = 4
+    
+    var floatingLabel = PaddedLabel()
+    var floatingLabelBottomConstraint:NSLayoutConstraint?
+    let floatingLabelBottomPaddingNormal:CGFloat = 33
+    let floatingLabelBottomPaddingActive:CGFloat = 8
+    let floatingLabelColorNormal:UIColor = UIColor.appColors.bluishGray
+    var floatingLabelColorActive:UIColor = UIColor.appColors.decredBlue
+    
+    var button = UIButton(type: .custom)
+    
+    var isActive:Bool = false
+    let padding = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 0)
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.initView()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.initView()
+    }
+    
+    private func initView() {
+        border.cornerRadius = self.cornerRadius
+        border.borderColor = self.inactiveBorderColor.cgColor
+        border.borderWidth = self.borderWidth
+        self.layer.addSublayer(border)
+        
+        self.initFloatedLabel()
+        self.addTarget(self, action: #selector(self.editingBegan), for: .editingDidBegin)
+        self.addTarget(self, action: #selector(self.editingEnded), for: .editingDidEnd)
+    }
+    
+    private func initFloatedLabel() {
+        self.floatingLabel.backgroundColor = UIColor.white
+        self.floatingLabel.textColor = self.floatingLabelColorNormal
+        self.floatingLabel.font = UIFont(name: self.font!.familyName, size: 16)
+        self.floatingLabel.clipsToBounds = true
+        self.floatingLabel.translatesAutoresizingMaskIntoConstraints = false
+        self.floatingLabel.sizeToFit()
+        self.floatingLabel.leftPadding = 4
+        self.floatingLabel.rightPadding = 4
+        self.floatingLabel.text = self.placeholder
+        self.placeholder = ""
+        self.addSubview(self.floatingLabel)
+        self.floatingLabelBottomConstraint = self.floatingLabel.bottomAnchor.constraint(equalTo: self.topAnchor, constant: self.floatingLabelBottomPaddingNormal)
+        NSLayoutConstraint.activate([
+            self.floatingLabel.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 8),
+            self.floatingLabelBottomConstraint!
+        ])
+    }
+    
+    func setPlaceHolder(text:String) {
+        self.floatingLabel.text = text
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        border.frame = self.bounds
+    }
+    
+    @objc func editingBegan() {
+        self.isActive = true
+        border.borderColor = self.activeBorderColor.cgColor
+        self.showFloatingLabel()
+    }
+    
+    @objc func editingEnded() {
+        self.isActive = false
+        border.borderColor = self.inactiveBorderColor.cgColor
+        self.hideFloatingLabel()
+    }
+    
+    func showError() {
```

Comment:
The `showFloatingLabel()` and `hideFloatingLabel()` functions should come before this.